activerecord/test/cases/associations/has_many_associations_test.rb

```diff
@@ -438,6 +438,28 @@ def test_create_resets_cached_counters
     assert_equal person, person.readers.first.person
   end
 
+  def test_update_all_respects_association_scope
+    person = Person.new
+    person.first_name = 'Naruto'
+    person.references << Reference.new
+    person.id = 10
+    person.references
+    person.save!
+    assert_nothing_raised do
+      person.references.update_all(favourite: true)
```

Comment:
Can we assert the return of `update_all` instead of using `assert_nothing_raised`?